tests/integration/devfile/cmd_devfile_debug_test.go

```diff
@@ -31,12 +31,13 @@ var _ = Describe("odo devfile debug command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("odo debug on a nodejs:latest component", func() {
-		It("check that machine output debug information works", func() {
+	When("creating a project using devfile with debugrun", func() {
```

Comment:
Since all the tests require creating a component with devfile-with-debugrun.yaml, it seems to me that we can put them all in a common `When("creating a component using devfile with debugrun", func(){})` block, and nest a `When("the component is pushed", func(){})` within it. There might some minor differences where one creation requires passing a component name but that shouldn't be a problem, we can generalize that for all the tests, i.e. use component name for all the creations.